### PR TITLE
Action Docs part2

### PR DESCRIPTION
**Pull request by chriselion**
### Proposed change(s)
Try to avoid using "vector action" or "{continuous, discrete} action space", and clean up a bit more around hybrid actions (without actually calling them that!). "action space" is still occasionally used when it's talking in the abstract, as opposed to specific continuous vs discrete.

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [x] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [x] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: docs/Learning-Environment-Design-Agents.md [hunk 1]
```diff
@@ -521,36 +521,34 @@ over time and many training episodes. Thus, the only place actions are defined
...
-with values ranging from zero to one.
+with values ranging from zero to one. You could even use a combination of continuous
+and discrete actions, for example using one continuous action for horizontal movement,
+and a discrete branch of size two for the vertical movement.
 
 Note that when you are programming actions for an agent, it is often helpful to
 test your action logic using the `Heuristic()` method of the Agent, which lets
 you map keyboard commands to actions.
 
-The [3DBall](Learning-Environment-Examples.md#3dball-3d-balance-ball) and
```
Comment (chriselion) [inline docs/Learning-Environment-Design-Agents.md:538]: Pretty sure this is no longer true.

File: docs/Getting-Started.md [hunk 2]
```diff
@@ -159,9 +159,6 @@ Unity brain name: 3DBallLearning
         Number of Visual Observations (per agent): 0
         Vector Observation space size (per agent): 8
         Number of stacked Vector Observation: 1
-        Vector Action space type: continuous
```
Comment (andrewcoh) [inline docs/Getting-Started.md:162]: Will there be no replacement here?

File: docs/Learning-Environment-Design-Agents.md [hunk 3]
```diff
@@ -83,7 +83,7 @@ As a concrete example, here is how the Ball3DAgent class implements these method
   method calls `VectorSensor.AddObservation()` such that vector size adds up to 8,
   the Behavior Parameters of the Agent are set with vector observation space
   with a state size of 8.
-- `Agent.OnActionReceived()` — The vector action spaces result
+- `Agent.OnActionReceived()` — The actions result
```
Comment (andrewcoh) [inline docs/Learning-Environment-Design-Agents.md:86]:
```suggestion
- `Agent.OnActionReceived()` — The action results
```

File: docs/Learning-Environment-Design-Agents.md [hunk 4]
```diff
@@ -511,7 +511,7 @@ setting the State Size.
 
 An action is an instruction from the Policy that the agent carries out. The
 action is passed to the Agent as the `ActionBuffers` parameter when the Academy invokes the
-agent's `OnActionReceived()` function. There are two types of actions supported:
+agent's `OnActionReceived()` function. There are two types of actions than an Agent can use:
```
Comment (andrewcoh) [inline docs/Learning-Environment-Design-Agents.md:514]:
```suggestion
agent's `OnActionReceived()` function. There are two types of actions that an Agent can use:
```

File: docs/Learning-Environment-Design-Agents.md [hunk 5]
```diff
@@ -521,36 +521,34 @@ over time and many training episodes. Thus, the only place actions are defined
 for an Agent is in the `OnActionReceived()` function.
 
 For example, if you designed an agent to move in two dimensions, you could use
-either continuous or the discrete vector actions. In the continuous case, you
-would set the vector action size to two (one for each dimension), and the
+either continuous or the discrete actions. In the continuous case, you
+would set the action size to two (one for each dimension), and the
 agent's Policy would create an action with two floating point values. In the
```
Comment (andrewcoh) [inline docs/Learning-Environment-Design-Agents.md:526]:
```suggestion
agent's Policy would output an action with two floating point values. In the
```

File: docs/Learning-Environment-Design-Agents.md [hunk 6]
```diff
@@ -521,36 +521,34 @@ over time and many training episodes. Thus, the only place actions are defined
...
 discrete case, you would use one Branch with a size of four (one for each
 direction), and the Policy would create an action array containing a single
 element with a value ranging from zero to three. Alternatively, you could create
 two branches of size two (one for horizontal movement and one for vertical
 movement), and the Policy would create an action array containing two elements
```
Comment (andrewcoh) [inline docs/Learning-Environment-Design-Agents.md:531]:
```suggestion
movement), and the Policy would output an action array containing two elements
```

File: docs/Learning-Environment-Design-Agents.md [hunk 7]
```diff
@@ -521,36 +521,34 @@ over time and many training episodes. Thus, the only place actions are defined
...
-with values ranging from zero to one.
+with values ranging from zero to one. You could even use a combination of continuous
+and discrete actions, for example using one continuous action for horizontal movement,
```
Comment (andrewcoh) [inline docs/Learning-Environment-Design-Agents.md:533]:
```suggestion
with values ranging from zero to one. You could alternatively use a combination of continuous
and discrete actions e.g., using one continuous action for horizontal movement
```

File: docs/Learning-Environment-Design-Agents.md [hunk 8]
```diff
@@ -570,15 +568,15 @@ These control values are applied as torques to the bodies making up the arm:
...
 these as well, if you plan to use a 3rd party algorithm with your environment.
 As shown above, you can scale the control values as needed after clamping them.
 
-### Discrete Action Space
+### Discrete Actions
 
 When an Agent's Policy uses **discrete** actions, the
 `ActionBuffers.DiscreteActions` passed to the Agent's `OnActionReceived()` function
-is an array of integers. When defining the discrete vector action space, `Branches`
+is an array of integers with length equal `Discrete Branch Size`. When defining the discrete actions, `Branches`
```
Comment (andrewcoh) [inline docs/Learning-Environment-Design-Agents.md:579]:
```suggestion
is an array of integers with length equal to `Discrete Branch Size`. When defining the discrete actions, `Branches`
```

File: docs/Learning-Environment-Design-Agents.md [hunk 9]
```diff
@@ -651,12 +649,12 @@ Notes:
...
 ### Actions Summary & Best Practices
 
-- Agents can either use `Discrete` or `Continuous` actions.
+- Agents can use `Discrete` and/or `Continuous` actions.
 - Discrete actions can have multiple action branches, and it's possible to mask
   certain actions so that they won't be taken.
-- In general, smaller action spaces will make for easier learning.
-- Be sure to set the Vector Action's Space Size to the number of used Vector
-  Actions, and not greater, as doing the latter can interfere with the
+- In general, a smaller number of action will make for easier learning.
```
Comment (andrewcoh) [inline docs/Learning-Environment-Design-Agents.md:655]:
```suggestion
- In general, fewer actions will make for easier learning.
```

File: docs/Learning-Environment-Design-Agents.md [hunk 10]
```diff
@@ -651,12 +649,12 @@ Notes:
...
+- Be sure to set the Continuous Action Size and Discrete Branch Size to desired
```
Comment (andrewcoh) [inline docs/Learning-Environment-Design-Agents.md:656]:
```suggestion
- Be sure to set the Continuous Action Size and Discrete Branch Size to the desired
```

File: docs/Learning-Environment-Design-Agents.md [hunk 11]
```diff
@@ -798,10 +796,9 @@ be called independently of the `Max Step` property.
       effective size of the vector observation being passed to the Policy being:
       _Space Size_ x _Stacked Vectors_.
   - `Vector Action`
-    - `Space Type` - Corresponds to whether action vector contains a single
-      integer (Discrete) or a series of real-valued floats (Continuous).
-    - `Space Size` (Continuous) - Length of action vector.
-    - `Branches` (Discrete) - An array of integers, defines multiple concurrent
+    - `Continuous Action Size` - The number of continuous actions that the Agent
```
Comment (andrewcoh) [inline docs/Learning-Environment-Design-Agents.md:799]: Should we say 'concurrent continuous actions'

File: docs/Learning-Environment-Examples.md [hunk 12]
```diff
@@ -36,7 +36,7 @@ you would like to contribute environments, please see our
   - +1.0 for arriving at optimal state.
 - Behavior Parameters:
   - Vector Observation space: One variable corresponding to current state.
-  - Vector Action space: (Discrete) Two possible actions (Move left, move
+  - Vector Action space: 1 discrete action branch with 3 actions (Move left, do nothing, move
```
Comment (andrewcoh) [inline docs/Learning-Environment-Examples.md:39]: Vector Action Space

File: docs/Training-Configuration-File.md [hunk 13]
```diff
@@ -36,7 +36,7 @@ choice of the trainer (which we review on subsequent sections).
...
+| `hyperparameters -> batch_size`             | Number of experiences in each iteration of gradient descent. **This should always be multiple times smaller than `buffer_size`**. If you are using a continuous actions, this value should be large (in the order of 1000s). If you are using a discrete action space, this value should be smaller (in order of 10s). <br><br> Typical range: (Continuous - PPO): `512` - `5120`; (Continuous - SAC): `128` - `1024`; (Discrete, PPO & SAC): `32` - `512`.                                                                                                                                                                                                                                                               |
```
Comment (andrewcoh) [inline docs/Training-Configuration-File.md:39]:
```suggestion
| `hyperparameters -> batch_size`             | Number of experiences in each iteration of gradient descent. **This should always be multiple times smaller than `buffer_size`**. If you are using continuous actions, this value should be large (on the order of 1000s). If you are using a discrete action space, this value should be smaller (on order of 10s). <br><br> Typical range: (Continuous - PPO): `512` - `5120`; (Continuous - SAC): `128` - `1024`; (Discrete, PPO & SAC): `32` - `512`.                                                                                                                                                                                                                                                               |
```

**Review by andrewcoh — APPROVED** [review]

File: docs/Getting-Started.md (hunk 2 above)
Comment (chriselion) [reply docs/Getting-Started.md:162]: Was waiting until we agreed on a rename. Will do as part of this PR.

Comment (chriselion) [reply docs/Getting-Started.md:162]: Oh wait, this is just the example log. It's all obsolete anyway.

File: docs/Learning-Environment-Examples.md (hunk 12 above)
Comment (chriselion) [reply docs/Learning-Environment-Examples.md:39]: Will rename all of these based on discussion of UI changes.

File: docs/Learning-Environment-Executable.md [hunk 14]
```diff
@@ -133,9 +133,6 @@ Unity brain name: Ball3DLearning
         Number of Visual Observations (per agent): 0
         Vector Observation space size (per agent): 8
         Number of stacked Vector Observation: 1
-        Vector Action space type: continuous
```
Comment (chriselion) [inline docs/Learning-Environment-Executable.md:136]: More console logs removed

File: docs/Learning-Environment-Design-Agents.md (hunk 11 above)
Comment (chriselion) [reply docs/Learning-Environment-Design-Agents.md:799]: Updated

**Comment by chriselion** [comment]
Remaining yamato tests aren't needed since none of the python changes affect runtime.

**Pull request merged**